File: Server/Hero.Server.Core/Models/Attribute.cs
namespace Hero.Server.Core.Models
{
    public class Attribute
    {
        public Guid Id { get; set; }
        public Guid? GroupId { get; set; }
        public string Name { get; set; }
        public string? IconData { get; set; }
        public string? Description { get; set; }
        public string? Category { get; set; }
        public double StepSize { get; set; }
        public double MinValue { get; set; }
        public double MaxValue { get; set; }

        public List<AttributeRace> AttributeRaces { get; set; }
        public List<AttributeSkill> AttributeSkills { get; set; }
        public void Update(Attribute attribute)
        {
            this.Name = attribute.Name;
            this.IconData = attribute.IconData;
            this.Description = attribute.Description;
            this.Category = attribute.Category;
            this.StepSize = attribute.StepSize;
            this.MinValue = attribute.MinValue;
            this.MaxValue = attribute.MaxValue;
            this.AttributeRaces = attribute.AttributeRaces;
            this.AttributeSkills = attribute.AttributeSkills;
        }
    }
}

File: Client/lib/src/features/character_traits/domain/attribute.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:kalinar/src/features/character_traits/domain/suggestable.dart';

import 'suggestion_state.dart';

part 'attribute.freezed.dart';
part 'attribute.g.dart';

@freezed
class Attribute extends Suggestable with _$Attribute {
  factory Attribute(
    String id,
    String groupId,
    String creatorId,
    String name,
    String? iconData,
    String? description,
    String? category,
    double stepSize,
    double minValue,
    double maxValue,
    SuggestionState state,
    String? rejectionReason,
    DateTime createdAt,
    DateTime? modifiedAt,
    DateTime? approvedAt,
    DateTime? rejectedAt,
  ) = _Attribute;

  factory Attribute.fromJson(Map<String, dynamic> json) => _$AttributeFromJson(json);
}
